src/lib.rs

```diff
@@ -145,7 +163,7 @@ pub struct Printer<'a, Writer: Write> {
     /// The buffered line built with each byte, ready to print to writer.
     buffer_line: Vec<u8>,
     writer: &'a mut Writer,
-    show_color: bool,
+    color_mode: Option<ColorMode>,
```

Comment:
Not necessary to change this now, but I was wondering if the code would improve here and there if we were to introduce `ColorMode::NoColor` instead of wrapping `ColorMode` inside an `Option<…>`.